webapp/lib/controller.dart

```diff
@@ -144,11 +144,7 @@ void initUI() {
 
   view.contentView.setUrlFilters(selectedTab, selectedProject, selectedPeriodFilter);
 
-  listenForNeedsReplyMetrics(selectedProject);
-  listenForDriverMetrics(selectedProject, DRIVERS);
-  listenForSystemEvents(PROJECTS);
-  listenForSystemMetrics();
-  // listenForDirectoryMetrics(); // not yet in use
+  command(UIAction.tabSwitched, null);
```

Comment:
instead of passing in `null` and then checking if the data is `null` in the command function, I'd recommend wrapping the `selectedTab` in a `ChartTypeData` object and passing it to this function call here, with a comment that `selectedTab` is already initialised at this point.

(I found the newly added if statement under `UIAction.tabSwitched` that tries to account for the data being null from this function call a bit confusing and was left wondering for a while why it wasn't throwing an error, until I read more of the code and realised that `selectedTab` gets initialised in `init`.)

As a general rule, making the values that get passed around explicit as much as possible makes it much easier to reason about how the code will behave.

Reply:
I agree. I've made the change